packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -228,31 +303,31 @@ class _CupertinoCheckboxState extends State<CupertinoCheckbox> with TickerProvid
           .withSaturation(_kCupertinoFocusColorSaturation)
           .toColor();
 
-    final Color effectiveCheckColor = widget.checkColor
-      ?? CupertinoColors.white;
-
     return Semantics(
       checked: widget.value ?? false,
       mixed: widget.tristate ? widget.value == null : null,
       child: buildToggleable(
         focusNode: widget.focusNode,
         autofocus: widget.autofocus,
-        onFocusChange: onFocusChange,
         size: const Size.square(kMinInteractiveDimensionCupertino),
         painter: _painter
+          ..position = position
+          ..reaction = reaction
           ..focusColor = effectiveFocusOverlayColor
-          ..isFocused = focused
           ..downPosition = downPosition
+          ..isFocused = states.contains(WidgetState.focused)
+          ..isHovered = states.contains(WidgetState.hovered)
           ..activeColor = effectiveActiveColor
-          ..inactiveColor = effectiveInactiveColor
-          ..checkColor = effectiveCheckColor
+          ..inactiveColor = _defaultFillColor.resolve(inactiveStates)
+          ..checkColor = _defaultCheckColor.resolve(states)
```

Comment:
The `states` is called for many times during the `build`, but each `states` actually makes an unnecessary new copy (unfortunately). Can we extract a `final currentStates = states;` at the beginning and use this one for read-only cases?